File: 383-ransom-note/ransom-note.c
bool canConstruct(char* ransomNote, char* magazine) {
    // 透過record記錄 magazine裡各個字元出現次數
    int record[26] = {0};
    for (int i = 0; i < strlen(magazine); i++) {
        record[magazine[i] - 'a']++;
    }
    for (int i = 0; i < strlen(ransomNote); i++) {
        if (record[ransomNote[i] - 'a'] <= 0) {
            return false;
        }else {
            record[ransomNote[i] - 'a']--;
        }
    }
    return true;
}

File: 383-ransom-note/ransom-note.go
func canConstruct(ransomNote string, magazine string) bool {
    if (len(magazine) < len(ransomNote)) {
        return false
    }
    record := make([]int, 26)
    for _, c := range magazine {
        record[c - 'a']++
    }
    for _, c := range ransomNote {
        if record[c - 'a'] <= 0 {
            return false
        } else {
            record[c - 'a']-=1
        }
    }
    return true
}